### PR TITLE
[w5.3b][W13-A1]Dominic Kenn Lim

### PR DESCRIPTION
**Pull request by dominickenn**
Extract commonalities from Phone, Email, Address to Contact class

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:3]: This class and its non-trivial methods should have header comments. 
If not, developers who write client code for this class will not know how to use it.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    private boolean isPrivate;
+    public String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:6]: Try to make it private. 
In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        return isPrivate;
+    }
+
+
+    public void setPrivacy(boolean isPrivate){
+        this.isPrivate = isPrivate;
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:23]: In general, `equals()` and `hashCode()` methods should be overridden together. Find out why!

File: src/seedu/addressbook/ui/Formatter.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
+package seedu.addressbook.ui;
+
+import java.util.List;
+
+import static seedu.addressbook.common.Messages.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/ui/Formatter.java:5]: Don't import everything using *;

**Comment by okkhoy** [comment]
@dominickenn 
This is quite a messy PR.
Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

Some comments added. Please ack & close the PR after reading comments.

**Comment by dominickenn** [comment]
Acknowledged.

**Pull request closed without merging**